nrf_modem/doc/rai.rst

```diff
@@ -0,0 +1,162 @@
+               .s_addr = nrf_htonl(NRF_INADDR_ANY)
+           }
+       };
+
+       /* Set the RAI_LAST option to indicate that no further data will be sent after this packet */
+       int option = NRF_RAI_LAST;
+       err = nrf_setsockopt(fd, NRF_SOL_SOCKET, NRF_SO_RAI, &option, sizeof(option));
+       if (err) {
+           printf("Failed to set NRF_SO_RAI option, error: %d\n", err);
+           return err;
+       }
+
+       /* Transmit the message */
+       err = nrf_sendto(fd, message, sizeof(message), 0,
+                        (struct nrf_sockaddr *)&dest_addr, sizeof(dest_addr));
+       if (err < 0) {
+           printf("Failed to send UDP packet with RAI_LAST, error: %d\n", err);
+           return err;
+       }
+
+       return 0;
+   }
+
+Example: Receiving a UDP Packet and Setting RAI_NO_DATA
+-------------------------------------------------------
+
+.. code-block:: c
+
+   #include <nrf_socket.h>
+
+   int receive_udp_packet_and_set_rai_no_data(int fd)
+   {
+       int err;
+       char buffer[128];
+       struct nrf_sockaddr_in src_addr;
+       nrf_socklen_t addrlen = sizeof(src_addr);
+
+       /* Receive an incoming message */
+       err = nrf_recvfrom(fd, buffer, sizeof(buffer), 0,
+                          (struct nrf_sockaddr *)&src_addr, &addrlen);
+       if (err < 0) {
+           printf("Failed to receive UDP packet, error: %d\n", err);
+           return err;
+       }
+
+       /* Set the RAI_NO_DATA option to signal that the device does not expect to send or receive further data */
+       int option = NRF_RAI_NO_DATA;
+       err = nrf_setsockopt(fd, NRF_SOL_SOCKET, NRF_SO_RAI, &option, sizeof(option));
+       if (err) {
+           printf("Failed to set NRF_SO_RAI option, error: %d\n", err);
+           return err;
+       }
+
+       return 0;
+   }
+
+.. note::
+
+   The examples assume network support for both CP-RAI and AS-RAI.
+
+.. note::
+
+   RAI flags can be used for secure sockets similarly to other sockets.
+
+.. note::
+
+   Implementing RAI should be a strategic decision based on the application's knowledge of its data transmission patterns. If there is uncertainty regarding future data transfers, it is advisable to refrain from using RAI. This caution helps avoid the additional energy expenditure associated with re-establishing the radio connection.
+
+Checking RAI Status
+*******************
+
+To verify if the RAI information was reported to the network:
+
+- For CP-RAI, check the ESM DATA TRANSPORT PDU sent by the modem.
+- For AS-RAI, currently, it is not visible in customer builds without specific modem traces.
+
+RAI Best Practices
+******************
+
+- Use RAI when no additional uplink or downlink traffic is expected in the near term.
+- Avoid using RAI if subsequent data transmission is anticipated, as it may lead to unnecessary RRC bearer reestablishments.
+- Consider each socket separately for RAI usage, and the modem will aggregate the RAI status across all sockets.
```

Comment:
```suggestion
Following are the best practices to be considered while using RAI:

* Use RAI when no additional uplink or downlink traffic is expected in the near term.
* Avoid using RAI if subsequent data transmission is anticipated, as it may lead to unnecessary RRC bearer reestablishments.
* Consider each socket separately for RAI usage, and the modem will aggregate the RAI status across all sockets.
```